apm-agent-plugins/apm-aws-sdk/apm-aws-sdk-1-plugin/src/test/java/co/elastic/apm/agent/awssdk/v1/DynamoDbClientIT.java

```diff
@@ -75,40 +74,63 @@ public void setupClient() {
     public void testDynamoDbClient() {
         Transaction transaction = startTestRootTransaction("s3-test");
 
-        executeTest("CreateTable", "query", TABLE_NAME, () -> dynamoDB.createTable(new CreateTableRequest().withTableName(TABLE_NAME)
-                .withAttributeDefinitions(List.of(
-                    new AttributeDefinition("attributeOne", ScalarAttributeType.S),
-                    new AttributeDefinition("attributeTwo", ScalarAttributeType.N)
-                ))
-                .withKeySchema(List.of(
-                    new KeySchemaElement("attributeOne", KeyType.HASH),
-                    new KeySchemaElement("attributeTwo", KeyType.RANGE)
-                ))
-                .withBillingMode(BillingMode.PAY_PER_REQUEST)),
-            dbAssert);
-
-        executeTest("ListTables", "query", null, () -> dynamoDB.listTables(),
-            dbAssert);
-
-        executeTest("PutItem", "query", TABLE_NAME, () -> dynamoDB.putItem(
+        newTest(() -> dynamoDB.createTable(new CreateTableRequest().withTableName(TABLE_NAME)
+            .withAttributeDefinitions(List.of(
+                new AttributeDefinition("attributeOne", ScalarAttributeType.S),
+                new AttributeDefinition("attributeTwo", ScalarAttributeType.N)
+            ))
+            .withKeySchema(List.of(
+                new KeySchemaElement("attributeOne", KeyType.HASH),
+                new KeySchemaElement("attributeTwo", KeyType.RANGE)
+            ))
+            .withBillingMode(BillingMode.PAY_PER_REQUEST)))
+            .operationName("CreateTable")
+            .entityName(TABLE_NAME)
+            .action("query")
+            .withSpanAssertions(dbAssert)
+            .execute();
```

Comment:
[for reviewer] this is mostly automatic refactoring with a "test builder" that allows to combine assertions on the spans in a more declarative way.